File: WavePlus/Communication/Packets/Outgoing/Roleplay/TargetLockComposer.cs
namespace Plus.Communication.Packets.Outgoing.Roleplay
{
    public class TargetLockComposer : MessageComposer
    {
        private readonly string _targetName;
        private readonly bool _isLocked;

        public TargetLockComposer(string targetName, bool isLocked)
            : base(ServerPacketHeader.TargetLockMessageComposer)
        {
            _targetName = targetName ?? string.Empty;
            _isLocked = isLocked;
        }

        public override void Compose(ServerPacket packet)
        {
            packet.WriteBoolean(_isLocked);
            packet.WriteString(_targetName);
        }
    }
}

File: WaveRP-Plugin/src/main/java/io/github/brenoepics/roleplay/communication/packets/emulator/outgoing/TargetLockComposer.java
package io.github.brenoepics.roleplay.communication.packets.emulator.outgoing;

import com.eu.habbo.messages.ServerMessage;
import com.eu.habbo.messages.outgoing.MessageComposer;

public class TargetLockComposer extends MessageComposer {

  private final String targetName;
  private final boolean isLocked;

  public TargetLockComposer(String targetName, boolean isLocked) {
    this.targetName = targetName;
    this.isLocked = isLocked;
  }

  @Override
  protected ServerMessage composeInternal() {
    this.response.init(6000);

    this.response.appendBoolean(this.isLocked);
    this.response.appendString(this.targetName);

    return this.response;
  }
}
